lint/rules/no-deprecated-icons.js

```diff
@@ -0,0 +1,60 @@
+module.exports = {
```

Comment:
really you need custom icon? We have rule for restricted imports

Reply:
you can have `import { Box, DeprecatedIcon } from '@macaw'` and you want warn only for `DeprecatedIcon` not `Box` itself - hence why I've added custom rule.